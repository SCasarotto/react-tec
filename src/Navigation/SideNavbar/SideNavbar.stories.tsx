import React from 'react';

import { Story, Meta } from '@storybook/react';

import { SideNavbar, SideNavbarProps } from './SideNavbar';

export default {
  title: 'Navigation/SideNavbar',
  component: SideNavbar,
  argTypes: {},
} as Meta;

const Template: Story<SideNavbarProps> = (args) => <SideNavbar {...args} />;

export const Default = Template.bind({});
Default.args = {};
