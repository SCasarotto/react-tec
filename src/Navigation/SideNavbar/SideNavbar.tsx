import React from 'react';

import { Link } from 'react-router-dom';

import { SideNavLink, SideNavLinkProps } from '../SideNavLink';
import { Container, Logo, NavContainer, MainUl } from './styledComponents';

export interface SideNavbarProps {
  sidebarWidth?: number;
  Header?: React.ReactNode;
  Footer?: React.ReactNode;
  logo?: string;
  links?: Array<SideNavLinkProps>;
  className?: string;
}
/**
 *
 * SideNavbar is the sidebar component you see to the left of this page.
 *
 */
export const SideNavbar: React.FC<SideNavbarProps> = (props) => {
  const {
    sidebarWidth = 200,
    logo,
    Header,
    links,
    className = '',
    Footer,
  } = props;
  return (
    <Container
      sidebarWidth={sidebarWidth}
      className={`tec-sideNavbar ${className}`}
    >
      {Header}
      {logo && !Header && (
        <Link to='/' className='tec-sideNavbarLogoLink'>
          <Logo src={logo} alt='logo' className='tec-sideNavbarLogo' />
        </Link>
      )}
      <NavContainer className='tec-sideNavbarNavContainer'>
        <MainUl className='tec-sideNavbarMainUl'>
          {links?.map((link) => {
            const { title, activePath, to, children, exact, key } = link;
            return (
              <SideNavLink
                key={key ?? to}
                to={to}
                activePath={activePath || to}
                title={title}
                exact={exact}
                className='tec-sideNavbarNavLink'
              >
                {children}
              </SideNavLink>
            );
          })}
        </MainUl>
      </NavContainer>
      {Footer}
    </Container>
  );
};
