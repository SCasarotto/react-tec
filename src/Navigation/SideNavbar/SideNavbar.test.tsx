import React from 'react';

import { render } from '@testing-library/react';

import { SideNavbar } from './SideNavbar';

describe('SideNavbar', () => {
  // Renders Without Error
  it('matches snapshot', () => {
    const { asFragment } = render(<SideNavbar />);
    expect(asFragment()).toMatchSnapshot();
  });

  // Class Name
  it('accepts custom className', () => {
    const { container } = render(<SideNavbar className='myClassName' />);
    expect(container.firstChild).toHaveClass('myClassName');
  });
});
