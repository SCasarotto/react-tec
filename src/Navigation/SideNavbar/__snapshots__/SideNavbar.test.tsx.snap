// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNavbar matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  position: fixed;
  top: 0px;
  bottom: 0px;
  left: 0px;
  max-height: 100vh;
  height: 100%;
  width: 200px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  z-index: 20;
  overflow-y: auto;
  box-shadow: rgba(0,0,0,0.1) 0px 0px 15px 0px, rgba(0,0,0,0.05) 1px 0px 5px 0px;
}

.c1 {
  height: 100%;
}

.c2 {
  margin: 0px;
  padding: 0px;
}

<div
    class="c0 tec-sideNavbar "
  >
    <div
      class="c1 tec-sideNavbarNavContainer"
    >
      <ul
        class="c2 tec-sideNavbarMainUl"
      />
    </div>
  </div>
</DocumentFragment>
`;
